### ISSUE TITLE
traps: vchroot[39952] general protection fault ip:7ffff80f7ce0 sp:7fffffdff290 error:0 in dyld[7ffff7fff000+153000]

### ISSUE THREAD

**Issue by HinTak** [darlinghq/darling#830]
Actually I just have this - and since I had darling up to date and working around 2nd/3rd June, I am suspecting the recent syscall related changes that got merged in the last 10 days. Unfortunately I am busy with other urgent things at the moment to test reverting those two changes.

@CuriousTommy. - Any thoughts?

**Comment by CuriousTommy** [comment]
Hmm, I see some macro mistakes in https://github.com/darlinghq/darling/commit/fc6a720edf065b51454d325472f75ec1456bb34e, let me fix those.

**Comment by HinTak** [comment]
@CuriousTommy I think my question was, did you see the vchroot trap before / after the change, and or recently?

**Comment by CuriousTommy** [comment]
Yeah, try to see if it works now.

On a side note, I wish I understood why eax needs to be positive for a 64bit call...

**Comment by HinTak** [comment]
@CuriousTommy - Meanwhile I have gotten round to reverting both of your recent syscall changes (actually I already reverted them when I filed this - just gotten round to build it, as it takes about 7 hours to do so...) - and indeed I got the older behavior back. So I take it reverting the 2nd one is important, even if we don't understand why yet.

**Comment by CuriousTommy** [comment]
https://github.com/darlinghq/darling/commit/fc6a720edf065b51454d325472f75ec1456bb34e should behave exactly the same as before on x86 systems. I designed this commit to use the older syscall if they exist. Plus I tested this on an x86 machine, so I know this commit should not prevent you from loading up the shell.

> So I take it reverting the 2nd one is important, even if we don't understand why yet.

It makes translating this x86 assembly code to ARM64 more annoying. If the negative check was done with `rax `, it would make a lot more sense.
